redefined flag "v" with k8s.io/klog/v2

**What happened**:

```
panic: /main flag redefined: v

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc000456070, {0x3bb7b18, 0x5a58500}, {0x3b8cd10, 0x1}, {0x35a2dfd, 0x14})
        /usr/local/go/src/flag/flag.go:1028 +0x37d
flag.IntVar(...)
        /usr/local/go/src/flag/flag.go:781
kubevirt.io/client-go/log.init.0()
        /src/vendor/kubevirt.io/client-go/log/log.go:88 +0x4d
```

kubevirt.io/client-go/log has defined command line flag "v" at [log/log.go](https://github.com/kubevirt/client-go/blob/babde8e0e236e3a75583ee889293e7d8d3041533/log/log.go#L88) , which was conflicted with [k8s.io/klog/v2](https://github.com/kubernetes/klog/blob/75663bb798999a49e3e4c0f2375ed5cca8164194/klog.go#L420)

**What you expected to happen**:
no confliction for any flags in  any direct/indirect imports.

**How to reproduce it (as minimally and precisely as possible)**:
1. import any packages under kubevirt.io/client-go contains this [commmit](https://github.com/kubevirt/client-go/commit/c26267d9b0b2922469ef0df502e2faad37204c28)
2. import any packages includes ki8s.io/klog/v2 

**Additional context**:

**Environment**:
- KubeVirt version (use `virtctl version`): v1.6.2
- Kubernetes version (use `kubectl version`): v1.32.5
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

So is it necessary for [this commit](https://github.com/kubevirt/client-go/commit/c26267d9b0b2922469ef0df502e2faad37204c28) to introduce an init() function to set default "v" flag?

--- a/cmd/fake-cmd-server/fake-cmd-server.go
+++ b/cmd/fake-cmd-server/fake-cmd-server.go
@@ -2,7 +2,6 @@ package main
 
 import (
 	"errors"
-	goflag "flag"
 
 	"github.com/spf13/pflag"
 	"go.uber.org/mock/gomock"
@@ -18,7 +17,7 @@ import (
 func main() {
 	socket := pflag.String("socket", cmdclient.SocketOnGuest(), "Socket for the cmd server")
 
-	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	pflag.CommandLine.AddGoFlag(log.VerbosityFlag())
 	pflag.Parse()
 
 	log.InitializeLogging("fake-cmd-server")

--- a/cmd/virt-launcher-monitor/virt-launcher-monitor.go
+++ b/cmd/virt-launcher-monitor/virt-launcher-monitor.go
@@ -22,7 +22,6 @@ package main
 import (
 	"bufio"
 	"errors"
-	goflag "flag"
 	"fmt"
 	"net/http"
 	"os"
@@ -62,7 +61,7 @@ func main() {
 	keepAfterFailure := pflag.Bool("keep-after-failure", false, "virt-launcher will be kept alive after failure for debugging if set to true")
 	uid := pflag.String("uid", "", "UID of the VirtualMachineInstance")
 
-	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	pflag.CommandLine.AddGoFlag(log.VerbosityFlag())
 	pflag.CommandLine.ParseErrorsWhitelist = pflag.ParseErrorsWhitelist{UnknownFlags: true}
 	pflag.Parse()
 

--- a/cmd/virt-launcher/virt-launcher.go
+++ b/cmd/virt-launcher/virt-launcher.go
@@ -22,7 +22,6 @@ package main
 import (
 	"context"
 	"errors"
-	goflag "flag"
 	"fmt"
 	"os"
 	"os/signal"
@@ -375,7 +374,7 @@ func main() {
 	libvirtLogFilters := pflag.String("libvirt-log-filters", "", "Set custom log filters for libvirt")
 	hypervisor := pflag.String("hypervisor", v1.KvmHypervisorName, "Hypervisor to be used by the VMI")
 
-	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	pflag.CommandLine.AddGoFlag(log.VerbosityFlag())
 	pflag.Parse()
 
 	log.InitializeLogging("virt-launcher")

--- a/cmd/virt-probe/virt-probe.go
+++ b/cmd/virt-probe/virt-probe.go
@@ -20,7 +20,6 @@
 package main
 
 import (
-	goflag "flag"
 	"fmt"
 	"os"
 	"runtime/pprof"
@@ -40,7 +39,7 @@ func main() {
 	timeoutSeconds := pflag.Int32("timeoutSeconds", 1, "Duration in seconds the probe will wait for the guest command to return.")
 	guestAgentPing := pflag.Bool("guestAgentPing", false, "Flag to specify readiness probe based of guest-agent ping")
 
-	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	pflag.CommandLine.AddGoFlag(log.VerbosityFlag())
 	pflag.Parse()
 
 	log.InitializeLogging("virt-probe")

--- a/cmd/virt-tail/main.go
+++ b/cmd/virt-tail/main.go
@@ -22,7 +22,6 @@ package main
 import (
 	"context"
 	"errors"
-	goflag "flag"
 	"fmt"
 	"os"
 	"os/signal"
@@ -102,7 +101,7 @@ func (v *VirtTail) tailLogs(location tail.SeekInfo) (*tail.SeekInfo, error) {
 }
 
 func main() {
-	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	pflag.CommandLine.AddGoFlag(log.VerbosityFlag())
 	pflag.CommandLine.ParseErrorsWhitelist = pflag.ParseErrorsWhitelist{UnknownFlags: true}
 	logFile := pflag.String("logfile", "", "path of the logfile to be streamed")
 	pflag.Parse()

--- a/pkg/service/BUILD.bazel
+++ b/pkg/service/BUILD.bazel
@@ -7,6 +7,7 @@ go_library(
     visibility = ["//visibility:public"],
     deps = [
         "//staging/src/kubevirt.io/client-go/kubecli:go_default_library",
+        "//staging/src/kubevirt.io/client-go/log:go_default_library",
         "//vendor/github.com/spf13/pflag:go_default_library",
     ],
 )

--- a/pkg/service/service.go
+++ b/pkg/service/service.go
@@ -27,6 +27,7 @@ import (
 	flag "github.com/spf13/pflag"
 
 	"kubevirt.io/client-go/kubecli"
+	kvlog "kubevirt.io/client-go/log"
 )
 
 func init() {
@@ -53,7 +54,7 @@ func (service *ServiceListen) Address() string {
 }
 
 func (service *ServiceListen) InitFlags() {
-	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	flag.CommandLine.AddGoFlag(kvlog.VerbosityFlag())
 	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("kubeconfig"))
 	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("master"))
 }

--- a/pkg/storage/export/virt-exportserver/exportserver.go
+++ b/pkg/storage/export/virt-exportserver/exportserver.go
@@ -26,7 +26,6 @@ import (
 	"crypto/x509"
 	"encoding/json"
 	"errors"
-	goflag "flag"
 	"fmt"
 	"io"
 	golog "log"
@@ -312,7 +311,7 @@ func (s *exportServer) buildServer(ctx context.Context) *http.Server {
 }
 
 func (s *exportServer) AddFlags() {
-	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	flag.CommandLine.AddGoFlag(log.VerbosityFlag())
 }
 
 func NewExportServer(config ExportServerConfig) (service.Service, error) {

--- a/pkg/virtctl/root.go
+++ b/pkg/virtctl/root.go
@@ -2,7 +2,6 @@ package virtctl
 
 import (
 	"context"
-	"flag"
 	"fmt"
 	"os"
 	"path/filepath"
@@ -144,14 +143,7 @@ func NewVirtctlCommandFn() *cobra.Command {
 }
 
 func addVerbosityFlag(fs *pflag.FlagSet) {
-	// The verbosity flag is added to the default flag set
-	// by init() in staging/src/kubevirt.io/client-go/log/log.go.
-	// We re-add it here to make it available in virtctl commands.
-	if f := flag.CommandLine.Lookup("v"); f != nil {
-		fs.AddFlag(pflag.PFlagFromGoFlag(f))
-	} else {
-		panic("failed to find verbosity flag \"v\" in default flag set")
-	}
+	fs.AddFlag(pflag.PFlagFromGoFlag(log.VerbosityFlag()))
 }
 
 func Execute() int {

--- a/staging/src/kubevirt.io/client-go/log/log.go
+++ b/staging/src/kubevirt.io/client-go/log/log.go
@@ -82,10 +82,29 @@ type FilteredLogger struct {
 
 var Log = DefaultLogger()
 
+var logFlags = goflag.NewFlagSet("kubevirt-log", goflag.ContinueOnError)
+
 func init() {
 	// "the practical default level is V(2)"
 	// see https://github.com/kubernetes/community/blob/master/contributors/devel/logging.md
-	goflag.IntVar(&defaultVerbosity, "v", 2, "log level for V logs")
+	logFlags.IntVar(&defaultVerbosity, "v", 2, "log level for V logs")
+}
+
+// VerbosityFlag returns the "-v" verbosity flag for bridging into
+// other flag sets (e.g. pflag). The flag is owned by a package-
+// internal FlagSet so that importing this package never registers
+// flags on flag.CommandLine and therefore cannot conflict with
+// other packages that register a "-v" flag there.
+//
+// Example:
+//
+//	pflag.CommandLine.AddGoFlag(log.VerbosityFlag())
+func VerbosityFlag() *goflag.Flag {
+	f := logFlags.Lookup("v")
+	if f == nil {
+		panic("kubevirt.io/client-go/log: verbosity flag \"v\" not registered on internal FlagSet")
+	}
+	return f
 }
 
 func InitializeLogging(comp string) {

--- a/staging/src/kubevirt.io/client-go/log/log_test.go
+++ b/staging/src/kubevirt.io/client-go/log/log_test.go
@@ -21,6 +21,7 @@ package log
 
 import (
 	"errors"
+	goflag "flag"
 	"fmt"
 	"path/filepath"
 	"runtime"
@@ -449,6 +450,40 @@ func TestObjectContextLeakage(t *testing.T) {
 		}
 	}
 
+	tearDown()
+}
+
+func TestVerbosityFlagDoesNotRegisterOnCommandLine(t *testing.T) {
+	goflag.CommandLine.VisitAll(func(f *goflag.Flag) {
+		if f.Name == "v" {
+			t.Fatal("\"v\" flag must not be registered on flag.CommandLine")
+		}
+	})
+}
+
+func TestVerbosityFlagReturnsNonNil(t *testing.T) {
+	f := VerbosityFlag()
+	if f == nil {
+		t.Fatal("VerbosityFlag() returned nil")
+	}
+	if f.Name != "v" {
+		t.Fatalf("VerbosityFlag() returned flag with name %q, want \"v\"", f.Name)
+	}
+}
+
+func TestVerbosityFlagSharesDefaultVerbosity(t *testing.T) {
+	setUp()
+	original := defaultVerbosity
+	defer func() { defaultVerbosity = original }()
+
+	f := VerbosityFlag()
+	if err := f.Value.Set("5"); err != nil {
+		t.Fatalf("Setting verbosity flag value: %v", err)
+	}
+	assert(t, defaultVerbosity == 5, "VerbosityFlag value must be shared with defaultVerbosity")
+
+	log := MakeLogger(MockLogger{})
+	assert(t, log.verbosityLevel == 5, "MakeLogger must pick up updated defaultVerbosity")
 	tearDown()
 }
 

